src/rqt_plot/plot_widget.py

```diff
@@ -154,6 +155,10 @@ def switch_data_plot_widget(self, data_plot):
         self.data_plot = data_plot
         self.data_plot_layout.addWidget(self.data_plot)
         self.data_plot.autoscroll(self.autoscroll_checkbox.isChecked())
+        if self.autoscale_checkbox.isChecked():
+            self.data_plot.set_autoscale(y=DataPlot.SCALE_EXTEND|DataPlot.SCALE_VISIBLE)
```

Comment:
The default value `SCALE_EXTEND|SCALE_VISIBLE` is currently being defined in the `plot.py` file. If the patch wants to make that toggle within the widget then the default value from the plot file should be removed and instead a function on the widget should be called to achieve the same behavior without duplicating these values.